infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -196,17 +203,39 @@
                     + " --version                                                                "
                     + "              Show version\n";
 
-    private static volatile Thread corfuServerThread;
-    private static volatile Thread shutdownThread;
+    // Active Corfu Server.
+    private static volatile CorfuServer activeServer;
+
+    // Flag if set to true - causes the Corfu Server to shutdown.
+    private static volatile boolean shutdownServer = false;
```

Comment:
Please don't use `static` variables. Instead of using a bunch of static variables, just create a new instance of  CorfuServer. Consider making CorfuServer singleton

Reply:
Done. please check #1871